apps/buildog/src/components/user-table.tsx

```diff
@@ -0,0 +1,199 @@
+  first_name: string;
+  last_name: string;
+  email: string;
+  created_at: string;
+  updated_at: string;
+};
+
+const dummyUsers: User[] = [
+  {
+    id: 1,
+    first_name: "John",
+    last_name: "Doe",
+    email: "john.doe@example.com",
+    created_at: "2023-07-15T10:15:30Z",
+    updated_at: "2023-09-10T14:20:45Z",
+  },
+  {
+    id: 2,
+    first_name: "Jane",
+    last_name: "Smith",
+    email: "jane.smith@example.com",
+    created_at: "2023-06-20T09:05:20Z",
+    updated_at: "2023-09-12T16:10:55Z",
+  },
+  {
+    id: 3,
+    first_name: "Michael",
+    last_name: "Brown",
+    email: "michael.brown@example.com",
+    created_at: "2023-08-01T11:25:40Z",
+    updated_at: "2023-09-11T18:30:35Z",
+  },
+  {
+    id: 4,
+    first_name: "Emily",
+    last_name: "Davis",
+    email: "emily.davis@example.com",
+    created_at: "2023-07-30T12:45:50Z",
+    updated_at: "2023-09-12T19:45:15Z",
+  },
+];
+
+export default function UserTable() {
+  const [users, setUsers] = useState<User[]>(dummyUsers);
+  const [open, setOpen] = useState(false);
+
+  const deleteUser = (userId: number) => {
+    setOpen(!open);
+
+    setUsers((prevUsers) => {
+      const updatedUsers = prevUsers.filter((user) => user.id !== userId);
+      return updatedUsers;
+    });
+  };
+
+  const tableData = () => {
+    return users.map((user) => {
+      const date = new Date(user.created_at);
+
+      const formattedDate = date.toLocaleDateString("en-GB");
+      const formattedTime = date.toLocaleTimeString("en-GB", {
+        hour: "2-digit",
+        minute: "2-digit",
+      });
+
+      return {
+        id: user.id,
+        first_name: user.first_name,
+        last_name: user.last_name,
+        email: user.email,
+        created_at: `${formattedDate} - ${formattedTime}`, // Combine date and time
+      };
+    });
+  };
+
+  const columns = [
+    {
+      id: "first_name",
+      accessorKey: "first_name",
+      header: "First Name",
+      enableSorting: true,
+    },
+    {
+      accessorKey: "last_name",
+      header: "Last Name",
+      enableSorting: false,
+      cell: ({ getValue }) => <div>{getValue()}</div>,
+    },
+    {
+      accessorKey: "email",
+      header: "E-Mail",
+      enableSorting: true,
+      cell: ({ getValue }) => <div>{getValue()}</div>,
+    },
+    {
+      accessorKey: "created_at",
+      header: "Creation Date",
+      enableSorting: true,
+      cell: ({ getValue }) => <div>{getValue()}</div>,
+    },
+    {
+      accessorKey: "actions",
+      header: "Actions",
+      id: "actions",
+      enableSorting: false,
+      cell: ({ row }) => {
+        const userId = row.original.id;
+        return (
+          <DropdownMenu>
+            <DropdownMenuTrigger asChild>
+              <Button variant="ghost" className="h-8 w-8 p-0">
+                ...
+              </Button>
+            </DropdownMenuTrigger>
+            <DropdownMenuContent align="end">
+              <DropdownMenuItem>Edit</DropdownMenuItem>
```

Comment:
What do you think about edit button? What is the flow for this?

question: Should we open a modal or should we change on the table with <select> after edit?